app/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source.rb

```diff
@@ -85,22 +87,51 @@ def find_templates_in_git_repo
               .group_by          { |file| File.dirname(file) }
               .select            { |_dir, files| files.any? { |f| f.end_with?(".tf", ".tf.json") } }
               .transform_values! { |files| files.map { |f| File.basename(f) } }
-              .each do |parent_dir, files|
-        name = self.class.template_name_from_git_repo_url(git_repository.url, parent_dir)
+              .each do |relative_path, files|
+        name = self.class.template_name_from_git_repo_url(git_repository.url, relative_path)
+
+        template_full_path = File.join(git_checkout_tempdir, relative_path)
 
-        # TODO: add parsing for input/output vars
-        input_vars  = nil
-        output_vars = nil
+        input_vars, output_vars, terraform_version = parse_vars_in_template(template_full_path)
 
         template_dirs[name] = {
-          :relative_path => parent_dir,
-          :files         => files,
-          :input_vars    => input_vars,
-          :output_vars   => output_vars
+          :relative_path     => relative_path,
+          :files             => files,
+          :input_vars        => input_vars,
+          :output_vars       => output_vars,
+          :terraform_version => terraform_version,
         }
       end
     end
 
     template_dirs
+  rescue => error
+    _log.error("Failing scaning for terraform templates in the git repo: #{error}")
+    raise error
+  ensure
+    cleanup_git_repo(git_checkout_tempdir)
+  end
+
+  # Parse template and return input-vars, output-vars & terraform-version
+  def parse_vars_in_template(template_path)
+    response = Terraform::Runner.parse_template_variables(template_path)
+    return response.template_input_params, response.template_output_params, response.terraform_version
+  end
+
+  # checkout git repo to temp dir
+  def checkout_git_repo
+    git_checkout_tempdir = Dir.mktmpdir("embedded-terraform-runner-git")
+
+    _log.debug("Checking out git repository to #{git_checkout_tempdir}...")
+    checkout_git_repository(git_checkout_tempdir)
+    git_checkout_tempdir
+  end
```

Comment:
I'm surprised there isn't a way to do this in the base `EmbeddedAutomationManager::ConfigurationScript` class, ideally we could have a `with_git_repository` which took a block and automatically handled the tempdir creation and cleanup.

Reply:
ok,  let me do this another PR,  rather than is PR.